### ISSUE TITLE
Better clarity for bundle items that have been included twice in the same bundle in different packages / add a non-packaged bundle tag or similar

### ISSUE THREAD

**Issue by Tecfan** [bartervg/barter.vg#179]
# What problem does this feature address?
Take this DLC: https://barter.vg/i/5793/

It was included in Humble Codemasters Bundle 2015 BOTH as a standalone DLC (tier 1) AND in the tier 3 package "GRID 2 All In DLC Pack".

When an item is included in a package, the following tag gets added on the /i/ page bundle section:
![image](https://user-images.githubusercontent.com/956878/92327403-db496f80-f059-11ea-830b-57d2ec3bdd6d.png)

This tag discourage people who are looking for the standalone item from even opening the bundle page to check the contents. If they did actually open the 2015 bundle, they would find that the "package" tag is _very_ inaccurate (something I found out a couple of weeks later).

## Describe a solution
We need better clarity for these types of situations. I don't know how the backend system works for bundle tags, but I assume a "non-packaged" bundle tag needs to be added. Nothing on the item page indicates that it was included twice either. This is also an issue in bundles where you get multiple copies, so a "multiple copies" tag is maybe also needed, see bundles like https://barter.vg/bundle/3409/ and the lack of "x2"/"x3" on the individual item pages.

## Examples of similar features
Bundle tags.

**Comment by Tecfan** [comment]
A second example just came into existence, 

Samorost 2 is now both in a package and available as a standalone key in Humble Indie Bundle 2. https://barter.vg/i/3513/

This is not possible to see from the /i/ page:

![image](https://user-images.githubusercontent.com/956878/93881319-092be680-fcdf-11ea-951a-6180bde35b0c.png)

**Comment by bartervg** [comment]
Fix could address this as well #126

**Comment by Tecfan** [comment]
New example: Peggle Deluxe is now in 3 different packages in the same bundle, but this is not noted on the game page:
![image](https://user-images.githubusercontent.com/956878/119251656-f63c1f80-bba7-11eb-8376-92c57c755c64.png)

(what decides which package is linked?) 

https://barter.vg/i/10148/
https://barter.vg/bundle/214/

**Comment by Tecfan** [comment]
See also this post: https://steamcommunity.com/groups/bartervg/discussions/1/1488866813777614422/?tscn=1634075550#c3082142648924997749

The example being: https://barter.vg/i/1368/

![image](https://user-images.githubusercontent.com/956878/137108712-bbe579e3-3d2c-4319-a7a3-aadde44d2c7d.png)

The game is in both tier 3, and in a package in tier 4. 

Barter currently reports "tier 3 of 4", + a package tag. While this is techincally correct, it is also hard to decipher without further research, so more clarity on the game page would be preferred

**Comment by Tecfan** [comment]
This is again an issue with the Company of Heroes Complete bundle.

https://barter.vg/i/156/

CoH base game listed as a Package, even though it is (also, and foremost) a standalone key.

**Comment by bartervg** [comment]
The design flaw was the assumption that item-bundle relationships would be unique. When an item is in a bundle more than once, it violates this assumption. These issues are some of the side effects.

Rather than add new tags, I reindexed item-bundle listings on a unique value. This avoids multiple occurrences merged into a single line. The advantage of this change is that the tier information is shown (e.g. tier 1 and tier 3). The disadvantage is that, at first glance, it may appear to be a duplication bug. In addition, bundles appear multiple times where it is not useful. 

![image](https://user-images.githubusercontent.com/28576132/208248704-d97dc3ca-3636-4ffc-ac87-e6e8e364e0a0.png)

This is fix is incomplete. The bundle tag system continues to rely on item-bundle to be unique. For example, the `400736 subID` tag on the tier 1 Company of Heroes is shown on the tier 2 package as well. The latter tag should not appear. Standalone and package has an easier fix, but the better fundamental fix would be ensure uniqueness on bundle tags.

![image](https://user-images.githubusercontent.com/28576132/208248466-9e84c6c9-04fc-4803-98f0-b1d2c447ee52.png)

The bundle count for Company of Heroes is 8 on the item profile page, 7 on the bundle page. Which is more accurate? It's appeared only once as a standalone game (in Yogcast 2019).

**Comment by bartervg** [comment]
> (what decides which package is linked?)

It was random. It changed on each pageview.

I added order by tier, but then saw that Peggle Deluxe had 3 different packages in the same tier. To ensure the order would not fluctuate randomly, I added order by a unique ID. That ID is determined when added to the bundle. Therefore it essentially orders by when the item added to the bundle.

**Comment by bartervg** [comment]
> so a "multiple copies" tag is maybe also needed,

The `multiple copies` tag has been available on bundles, but is rarely used. I added it to https://barter.vg/bundle/3409/ and I can see why. The tag is easy to miss `⁘` and doesn't display the number of copies. Since it's in between other icons such as library and blacklist, it may be confusing to use `x` icon. 
![image](https://user-images.githubusercontent.com/28576132/208249450-b48de4ef-db0f-40cd-bea7-07f312b21d5f.png)

It's a bit better on the item profile pages. This page shows the number (e.g. number of copies, subID number) for all tags.
![image](https://user-images.githubusercontent.com/28576132/208249437-e47b270a-cdf1-466c-a8ca-d26c28a8915b.png)

Perhaps the extra copies tag should appear separately from other tags. 
![image](https://user-images.githubusercontent.com/28576132/208249555-5d0f166c-da0b-4bcc-974f-3a7184c96612.png)

**Comment by bartervg** [comment]
> In addition, bundles appear multiple times where it is not useful.

The bundles, which the related packages are in, no longer appear for the pulldown for the app. Since the app isn't in the bundle as a standalone item, it was misleading to list those bundles. 

![image](https://user-images.githubusercontent.com/28576132/208521593-79e94dc8-271c-43e4-8960-01e295986cd4.png)
 
The internal Barter focus group was strongly against the duplicate listings of bundles. The particularly bad example was Company of Heroes 2 with 8 rows for 3 bundles, 
![image](https://user-images.githubusercontent.com/28576132/208522162-ce61b475-bc1e-4ad8-b132-e8ac6ac98d34.png)
and now it is consolidated into 3 rows,
![image](https://user-images.githubusercontent.com/28576132/208522411-032563d3-6122-4f36-ae21-de6ef38f0f13.png)
 
> The bundle tag system continues to rely on item-bundle to be unique. 

This was a convoluted fix, but turned up many related errors. 
- Most of the bundle tags weren't visible because the format changed and the previous tags had 0 values, which were invalid and did not display. I set the user value to Barter Bot, since the original tagger value was never recorded, and the tags appear now.
- Items that do not exist yet can be added to bundles. They are invisible until the profile is created later. I deleted 6 rows, but there may have been previously-non-existent items added to bundles that the profile exist now, but do not belong in the bundles. For example https://barter.vg/i/198479/ exists (as of posting date), while https://barter.vg/i/298479/ does not. Attempting to input the latter no longer stealthily adds an invisible item to the bundle.
- If there were two identical items in a bundle, it wasn't possible to select one of them. This was particular annoying for bundle editors.

> the lack of "x2"/"x3" on the individual item pages.

The tag will appear on the item profile pages if someone applied it on the bundle. 

However, this remains a problem for https://barter.vg/i/117/#bundles and for bundles where the same item appears more than once (rather than app and package). The subsequent additions doesn't display tags on the item profile page. I checked most of the 321 items where items appear in a bundle more than once, and nearly all of them were LootBoy or lequeshop. Therefore, no plan to fix this on the item profile pages.
